src/pages/api/users/[id].js

```diff
@@ -0,0 +1,30 @@
+import dbConnect from "@/utils/dbConnect";
+import { deleteUser, modifyUser } from "@/utils/dbFunctions";
+
+export default async function handler(req, res) {
+  const { method } = req;
+
+  await dbConnect();
+
+  switch (method) {
+    case "DELETE":
```

Comment:
This is just a personal preference but I really don't like `switch` statements, and here's why:

**They are verbose**. An if statement would require less lines (1 instead of 2) for each branch, so even though it's easy to think that `switch` is a more concise way of achieving the same thing, it really takes up more space in the dimension where space is most restricted, i.e. the vertical. They also add extra indentation to the body of the code, e.g. lines 11-16 here, so the bulk of the code actually gets wider in the horizontal dimension as well. I bring this up first not because it's the most important shortcoming, but because an often cited reason to use `switch` is to save space.

**It's error-prone**. It's very easy to forget to add the `break` statement at the end of a branch, which will not get caught by syntax checking or by formatting. Omitting the `break` is actually considered a feature, called _fall-through_, and even though you very rarely want to use that feature, it's the default behaviour in most programming languages. So forgetting the `break` will cause hard-to-find bugs.

**It's one more tool to master, but less capable**. When you use `if`statements, you can use any condition that evaluates to true or false. With `switch`, your cases (in most situations) won't be boolean expressions but have to relate directly to the `switch` value. This is not a big problem on its own, because you will only use `switch` in situations that match these capabilities. But why have two tools, if one of them (`if … else`) can solve all problems, and the other can only solve a very narrow space of problems?

All in all, I recommend against using `switch`. But although I can argue my case, it mostly boils down to personal preference. 😊

Reply:
I'm a fan of switch statements when it's for instance comparing a variable to strings, like I've used it here. I think it adds some readability and looks better than `if (variable == "this")  do that else if (variable == "that") do that`
Appreciate your feedback though and I definitely see your reasoning!